elastic/beats
# Disable kubeadm for filebeat (both filbeat.input and filbeat.autodiscover)

- Enhancement

~NOTE: Dont merge until https://github.com/elastic/elastic-agent-autodiscover/pull/98 is also merged~

## Proposed commit message

**WHAT:** Provide a configuration option to disable /api/v1/namespaces/kube-system/configmaps/kubeadm-config for **filebeat autodiscovery** and **add_kubernetes_metadata processor**. In more details user will disable the api requests  by using configuration use_kubeadm: false

In this PR we added the option in order to support both **filbeat.input** and **filebeat.autodiscover** blocks of configuration.

**WHY:** https://github.com/elastic/enhancements/issues/21172

## Checklist





## How to test this PR locally

1. Follow same steps as described https://github.com/elastic/beats/pull/40086
2. The filebeat configuration should contain only autodiscover config: https://github.com/elastic/beats/blob/main/deploy/kubernetes/filebeat-kubernetes.yaml#L133C1-L150C42


## Related issues

- Relates #https://github.com/elastic/beats/pull/40086 
- Relates #https://github.com/elastic/beats/issues/40276



## Screenshots

![Screenshot 2024-09-10 at 11 27 21 AM](https://github.com/user-attachments/assets/ebaa723b-9c40-4a20-a9c9-3a7b12ebbf63)

LOG processing continues successfully:

![Screenshot 2024-09-10 at 11 51 22 AM](https://github.com/user-attachments/assets/e9f322f9-78cd-4824-9a5d-ea0d1596ffe2)


For the filebeat.input [block](https://github.com/elastic/beats/blob/main/deploy/kubernetes/filebeat-kubernetes.yaml):

![Screenshot 2024-09-10 at 2 17 25 PM](https://github.com/user-attachments/assets/e7b52095-c366-48ca-a8ef-04c3224ff982)

LOG processing continues successfully:
![Screenshot 2024-09-10 at 2 30 40 PM](https://github.com/user-attachments/assets/a9a5e99c-dcdd-4121-b13e-b5a1ab0c89c9)
